detector/zoo/plugin.go

```diff
@@ -6,6 +6,10 @@ import (
 
 func init() {
 	detector.Register("zk://", detector.PluginFactory(func(spec string) (detector.Master, error) {
-		return NewMasterDetector(spec)
+		md, err := NewMasterDetector(spec)
+		if err == nil {
+			go md.client.connect()
+		}
```

Comment:
In my branch, i tried to connect() in detect (with the Start() method), I ran into issues during testing with the connection channel reclosing, etc.  Could have been my mistake, but ended up calling connect() way earlier than in Detect().